Relax ruby version constraint.

alext opened the pull request:
## What

This was pining the exact ruby version. This resulted in a [deployment
failure](https://concourse.build.ci.cloudpipeline.digital/teams/main/pipelines/paas-product-page/jobs/deploy/builds/42) when the buildpack was updated and dropped support for 2.4.1

This therefore updates the version constraints to use pessimestic
version pinning so that it uses the latest 2.4.x supported by the
buildpack.

Specifying '2.4' in the .travis.yml causes Travis to pick the latest
2.4.x version it has available[1].

It's not possible to specify anything other than an exact ruby version
in the .ruby-version file, so this has been udpated to 2.4.3, which is
the latest 2.4 version at present.

[1]https://docs.travis-ci.com/user/languages/ruby/#Specifying-Ruby-versions-and-implementations

I've also fixed the bundler gem name in the `Dockerfile` because I spotted it at the same time.

## How to review

Code review and running locally is probably enough.

## Who can review

Not me.

alext committed:
Relax ruby version constraint.

This was pining the exact ruby version. This resulted in a deployment
failure when the buildpack was updated and dropped support for 2.4.1

This therefore updates the version constraints to use pessimestic
version pinning so that it uses the latest 2.4.x supported by the
buildpack.

Specifying '2.4' in the .travis.yml causes Travis to pick the latest
2.4.x version it has available[1].

It's not possible to specify anything other than an exact ruby version
in the .ruby-version file, so this has been udpated to 2.4.3, which is
the latest 2.4 version at present. I've also stopped reading this in the
Gemfile as it's no longer the source of truth (and it was breaking
Travis which only has 2.4.1 available)

[1]https://docs.travis-ci.com/user/languages/ruby/#Specifying-Ruby-versions-and-implementations

samcrang reviewed (approved):
I'll merge that when Travis passes.

alext committed:
Fix bundler gem name in Docker image.

bundle is a convenience gem[1] for people who typo the gem name when
installing bundler

[1]https://rubygems.org/gems/bundle
